Source/ASTextNode2.mm

```diff
@@ -228,18 +226,16 @@ - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
 {
   ASDisplayNodeAssert(constrainedSize.width >= 0, @"Constrained width for text (%f) is too  narrow", constrainedSize.width);
   ASDisplayNodeAssert(constrainedSize.height >= 0, @"Constrained height for text (%f) is too short", constrainedSize.height);
-
+  
   ASLockScopeSelf();
-
+  
+  [self _ensureTruncationText];
   ASTextContainer *container = [_textContainer copy];
-  NSAttributedString *attributedText = self.attributedText;
   container.size = constrainedSize;
-  [self _ensureTruncationText];
   
-  NSMutableAttributedString *mutableText = [attributedText mutableCopy];
-  [self prepareAttributedString:mutableText];
-  ASTextLayout *layout = [ASTextNode2 compatibleLayoutWithContainer:container text:mutableText];
+  ASTextLayout *layout = [self compatibleLayoutWithContainer:container text:[self l_preparedAttributedText]];
   
+  // Is this necessary?
   [self setNeedsDisplay];
```

Comment:
Hmm, I think you should remove this (as long as the tests pass, or fix them if the tests are not too hard to fix).

As long as we have the layer's .needsDisplayOnBoundsChange set to YES, then this should not be a factor.

One could argue that we should use default-synchronous display whenever the layout changes while interfaceState is Visible. That would fix some of the stretching texture issues that often happen when changing a string and it resizes the frame. However, this would require something like setNeedsSynchronousDisplay, and it would be called in -layout instead of here.